src/team-calendar-service.ts

```diff
@@ -26,20 +25,27 @@ export function serviceHandler(srv) {
       });
   });
 
-  srv.after("UPDATE", "Appointment", async ([appointment]: [Appointment]) => {
-    // split into multiple days and write in parallel
-    const person = await personById(srv, appointment.person_ID);
-
-    return Promise.all(
-      splitAppointmentIntoDays(appointment)
-        .map(day => buildTimeSheetEntry(appointment, person, day))
-        .map(writeTimeSheetEntry)
-    )
-    .then(() => {
-      return appointment;
-    })
-    .catch(error => {
-      throw Error(`Failed to create appointment! ${error.message}`);
-    });
+  srv.after("UPDATE", "Appointment", async (appointment: Appointment, req) => {
+    //the transaction joins the previous read request
+    const tx = srv.transaction(req);
+    const persons = await tx.run(SELECT.from ("Person").where({ID:appointment.person_ID}));
+    if (persons.length === 0) {
+      throw new Error(`No person found with ID ${appointment.person_ID}`);
+    }
+    else{
+      const [person] = persons;
```

Comment:
Actually, I would rather move that up, i. e.:

```ts
const [person] = await tx.run(SELECT...

if (!person) {
  throw Error...
}
```

Have you checked what happens if there is no user with a given ID? You can easily try this by just passing any string to the where clause above. I would expect that an error is thrown, but from the code here it seems that an empty array is returned?

Reply:
I checked tand the error is thrown in both cases, I'll apply your suggestion